### ISSUE TITLE
qmk_firmware Broken in NixOS 20.09

### ISSUE THREAD

**Issue by mcdviii** [NixOS/nixpkgs#117077]
**Describe the bug**
`qmk_firmware` package appears to be broken.

**To Reproduce**
1. Add `qmk_firmware` to `environment.systemPackages`
2. `nixos-rebuild switch` 
```
unpacking channels...
building Nix...
building the system configuration...
error: Package ‘qmk_firmware-0.6.144’ in /nix/store/fm0ckk0n6iyh6pm7m0f81l1brnf2blg2-nixos-20.09/nixos/pkgs/development/misc/qmk_firmware/default.nix:10 is marked as broken, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

(use '--show-trace' to show detailed location information)
```
3. `nixos-rebuild switch --show-trace`
```
building Nix...
building the system configuration...
error: while evaluating the attribute 'activationScript' of the derivation 'nixos-system-liber-20.09.3537.94166491939' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/activation/top-level.nix:95:5:
while evaluating the attribute 'system.activationScripts.script' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/activation/activation-script.nix:80:9:
while evaluating 'textClosureMap' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/strings-with-deps.nix:70:35, called from /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/activation/activation-script.nix:101:18:
while evaluating 'id' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/trivial.nix:14:5, called from undefined position:
while evaluating the attribute 'text' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/activation/activation-script.nix:9:5:
while evaluating the attribute 'text' at undefined position:
while evaluating 'g' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:276:19, called from undefined position:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:98:72, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:279:20:
while evaluating the attribute 'value' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:465:9:
while evaluating the option `system.activationScripts.etc.text':
while evaluating the attribute 'mergedValue' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:497:5:
while evaluating the attribute 'values' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:491:9:
while evaluating the attribute 'values' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:590:7:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:477:28, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:477:17:
while evaluating definitions from `/nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/etc/etc.nix':
while evaluating 'dischargeProperties' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:549:25, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:478:137:
while evaluating the attribute 'value' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:338:44:
while evaluating the attribute 'sources' of the derivation 'etc' at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/etc/etc.nix:12:5:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/system/etc/etc.nix:20:20, called from undefined position:
while evaluating the attribute 'source' at undefined position:
while evaluating 'g' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:276:19, called from undefined position:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:98:72, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/attrsets.nix:279:20:
while evaluating the attribute 'value' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:465:9:
while evaluating the option `environment.etc.dbus-1.source':
while evaluating the attribute 'mergedValue' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:497:5:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:499:17, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:499:12:
while evaluating 'check' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/types.nix:256:15, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:499:22:
while evaluating the attribute 'serviceDirectories' of the derivation 'dbus-1' at /nix/store/fm0ckk0n6iyh6pm7m0f81l1brnf2blg2-nixos-20.09/nixos/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/types.nix:265:14, called from undefined position:
while evaluating the attribute 'value' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:510:27:
while evaluating anonymous function at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:499:17, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:499:12:
while evaluating 'check' at /nix/var/nix/profiles/per-user/root/channels/nixos/lib/types.nix:256:15, called from /nix/var/nix/profiles/per-user/root/channels/nixos/lib/modules.nix:499:22:
while evaluating the attribute 'passAsFile' of the derivation 'system-path' at /nix/store/fm0ckk0n6iyh6pm7m0f81l1brnf2blg2-nixos-20.09/nixos/pkgs/build-support/trivial-builders.nix:7:7:
while evaluating the attribute 'handled' at /nix/store/fm0ckk0n6iyh6pm7m0f81l1brnf2blg2-nixos-20.09/nixos/pkgs/stdenv/generic/check-meta.nix:256:7:
while evaluating 'handleEvalIssue' at /nix/store/fm0ckk0n6iyh6pm7m0f81l1brnf2blg2-nixos-20.09/nixos/pkgs/stdenv/generic/check-meta.nix:152:38, called from /nix/store/fm0ckk0n6iyh6pm7m0f81l1brnf2blg2-nixos-20.09/nixos/pkgs/stdenv/generic/check-meta.nix:257:14:
Package ‘qmk_firmware-0.6.144’ in /nix/store/fm0ckk0n6iyh6pm7m0f81l1brnf2blg2-nixos-20.09/nixos/pkgs/development/misc/qmk_firmware/default.nix:10 is marked as broken, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.
```

**Notify maintainers**

No maintainers listed.

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/a9g7p6fwanw66j5djzila7ql1hky759z-bash-interactive-4.4-p23-dev
copying path '/nix/store/a9g7p6fwanw66j5djzila7ql1hky759z-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.105, NixOS, 20.09.3537.94166491939 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-20.09"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by turboMaCk** [comment]
Actual build errors are:

```
Making preonic/rev2 with keymap default                                                                [ERRORS]
make[1]: Entering directory '/build/source'

 * The firmware is too large! 29732/28672 (1060 bytes over)

make[1]: *** [tmk_core/rules.mk:379: check-size] Error 1
make[1]: Leaving directory '/build/source'
Making preonic/rev1 with keymap default                                                                [ERRORS]
make[1]: Entering directory '/build/source'

 * The firmware is too large! 29732/28672 (1060 bytes over)

make[1]: *** [tmk_core/rules.mk:379: check-size] Error 1
make[1]: Leaving directory '/build/source'
Making preonic/rev2 with keymap default                                                                [ERRORS]
make[1]: Entering directory '/build/source'

 * The firmware is too large! 29732/28672 (1060 bytes over)
```

**Comment by dotlambda** [comment]
See https://github.com/NixOS/nixpkgs/pull/109639.

**Comment by dotlambda** [comment]
@mcdviii This is unlikely to be fixed for 20.09. But maybe you can help @AndersonTorres to make it work on master.

**Comment by AndersonTorres** [comment]
Set `broken = true;` on stable. After all, it was effectively dropped on master ().
